### PR TITLE
#1155 Resource paging to assist Github/Gitlab resources.

### PR DESCRIPTION
**Pull request by criske**
pr for #1155

This will be used to implement `GitlabStars#added()`.

### CODE REVIEW

**Comment by zoeself** [comment]
@criske thank you for your Pull Request. I'll assign someone to review it soon.

**Comment by criske** [comment]
@amihaiemil Unfortunately there is no straight way the check if current user has starred "this" repo. The only endpoint is to get all starred repo by this user (https://docs.gitlab.com/ee/api/projects.html#list-projects-starred-by-a-user)
and then check if this repos is in that list.

Problem is that this requires pagination even though we pas "per_page =100". So this PR come with a Resource Pagination / Headers based implementation that could be used be implement "added()" and in the future for other cases, either for
Github or Gitlab (they use the same format).

**Comment by zoeself** [comment]
@amihaiemil I couldn't find any assignee for this task. This is either because there are no contributors with role ``REV`` available or because the project does not have enough funds.

Please, make sure there is at least one available contributor with the required role and the project can afford to pay them.

File: self-core-impl/src/main/java/com/selfxdsd/core/ResourcePaging.java [hunk 1]
```diff
@@ -0,0 +1,177 @@
...
+import java.util.stream.StreamSupport;
+
+/**
+ * Resource Pagination. Implementations should extract the next page link
+ * either from headers or from body, depending on the Provider's API.
+ * @author criske
+ * @version $Id$
+ * @since 0.0.84
+ */
+interface ResourcePaging extends Iterable<Resource> {
```
Comment (amihaiemil) [inline self-core-impl/src/main/java/com/selfxdsd/core/ResourcePaging.java:44]: @criske looks interesting, but it's not clear to me how this should be used. Can you extend the JavaDoc with usage examples? :D

**Review by amihaiemil — COMMENTED** [review]
@criske question

**Comment by amihaiemil** [comment]
@criske very elegant, I like it! Thanks! :D

File: self-core-impl/src/main/java/com/selfxdsd/core/ResourcePaging.java (hunk 1 above)
Comment (criske) [reply self-core-impl/src/main/java/com/selfxdsd/core/ResourcePaging.java:82]: @amihaiemil basically it implements this https://docs.github.com/en/rest/guides/traversing-with-pagination (same format is used by Gitlab too)

**Pull request merged**